specs/ContextMenuRequested.md

```diff
@@ -0,0 +1,711 @@
+            [in] ICoreWebView2CustomItemSelectedEventHandler* eventHandler,
+            [out] EventRegistrationToken* token);
+
+        /// Remove an event handler previously added with add_CustomItemSelected.
+        HRESULT remove_CustomItemSelected(
+            [in] EventRegistrationToken token);
+        
+    }
+
+    /// A continuation of the ICoreWebView2Environment interface.
+    [uuid(04d4fe1d-ab87-42fb-a898-da241d35b63c), object, pointer_default(unique)]
+    interface ICoreWebView2Environment : IUnknown
+    {
+        /// Create a ContextMenuItem object used for developers to insert new items into the default browser context menu
+        /// The Enabled and visible properties will default to true and the IsChecked property will only be used if the 
+        /// menu item type is radio or checkbox
+        HRESULT CreateContextMenuItem(
+            [in] UINT32 customItemID,
+            [in] LPCWSTR displayName,
+            [in] LPCWSTR shorctut,
+            [in] IStream* iconStream,
+            [in] COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND kind,
+            [in] BOOL isChecked,
+            [out, retval] ICoreWebView2ContextMenuItem ** item);
+    }
+
+    [uuid(04d3fe1d-ab87-42fb-a898-da241d35b63c), object, pointer_default(unique)]
+    interface ICoreWebView2ContextMenuRequestedEventHandler : IUnknown
+    {
+        /// Called to provide the event args when a context menu is invoked on WebView2 element
+        HRESULT Invoke(
+            [in] ICoreWebView2* sender,
+            [in] ICoreWebView2ContextMenuRequestedEventArgs* args);
+    }
+
+    [uuid(49e1d0bc-fe9e-4481-b7c2-32324aa21998), object, pointer_default(unique)]
+    interface ICoreWebView2CustomItemSelectedEventHandler : IUnknown
+    {
+        /// Called to provide the event args when the end user selects on an dev-inserted context menu item
+        HRESULT Invoke(
+            [in] ICoreWebView2* sender,
+            [in] ICoreWebView2CustomItemSelectedEventArgs* args);
+    }
+
+    /// Event args for the ContextMenuRequested event. Will contain the selection information and a collection of all of the default context menu items that the default WebView2 menu would show and allows the app to draw its own context menu or add/ remove from the default context menu.
+    [uuid(a1d309ee-c03f-11eb-8529-0242ac130003), object, pointer_default(unique)]
+    interface ICoreWebView2ContextMenuRequestedEventArgs : IUnknown
+    {
+        /// The list of default ContextMenuItem objects
+        [propget] HRESULT MenuItems([out, retval] ICoreWebView2ContextMenuItemCollection ** value);
+
+        /// Contains the data regarding the selection
+        [propget] HRESULT ContextMenuParams([out, retval] ICoreWebView2ContextMenuParams ** value);
+
+        /// Returns the selected Context Menu Item
+        [propget] HRESULT SelectedItem([out, retval] ICoreWebView2ContextMenuItem ** value);
+
+        /// Sets the Selected Menu Item for the browser to execute the command
```

Comment:
Please note in the documentation what the default value for this property is and what it means to set this value to null. I assume this value defaults to null and that null means there was no selection and the context menu was cancelled?